sycl/source/detail/builtins_helper.hpp

```diff
@@ -70,40 +73,42 @@
   }
 
 #define __MAKE_1V_2S(Fun, N, Ret, Arg1, Arg2)                                  \
-  __SYCL_EXPORT Ret##N Fun __NOEXC(Arg1##N x, Arg2 y) {                        \
-    Ret##N r;                                                                  \
+  __SYCL_EXPORT sycl::vec<Ret, N> Fun __NOEXC(sycl::vec<Arg1, N> x, Arg2 y) {  \
+    sycl::vec<Ret, N> r;                                                       \
     detail::helper<N - 1>().run_1v_2s(                                         \
         r, [](Arg1 x, Arg2 y) { return cl::__host_std::Fun(x, y); }, x, y);    \
     return r;                                                                  \
   }
 
 #define __MAKE_SR_1V_AND(Fun, Call, N, Ret, Arg1)                              \
-  __SYCL_EXPORT Ret Fun __NOEXC(Arg1##N x) {                                   \
+  __SYCL_EXPORT Ret Fun __NOEXC(sycl::vec<Arg1, N> x) {                        \
     Ret r;                                                                     \
     detail::helper<N - 1>().run_1v_sr_and(                                     \
         r, [](Arg1 x) { return cl::__host_std::Call(x); }, x);                 \
     return r;                                                                  \
   }
 
 #define __MAKE_SR_1V_OR(Fun, Call, N, Ret, Arg1)                               \
-  __SYCL_EXPORT Ret Fun __NOEXC(Arg1##N x) {                                   \
+  __SYCL_EXPORT Ret Fun __NOEXC(sycl::vec<Arg1, N> x) {                        \
     Ret r;                                                                     \
     detail::helper<N - 1>().run_1v_sr_or(                                      \
         r, [](Arg1 x) { return cl::__host_std::Call(x); }, x);                 \
     return r;                                                                  \
   }
 
 #define __MAKE_1V_2P(Fun, N, Ret, Arg1, Arg2)                                  \
-  __SYCL_EXPORT Ret##N Fun __NOEXC(Arg1##N x, Arg2##N *y) {                    \
-    Ret##N r;                                                                  \
+  __SYCL_EXPORT sycl::vec<Ret, N> Fun __NOEXC(sycl::vec<Arg1, N> x,            \
+                                              sycl::vec<Arg2, N> *y) {         \
+    sycl::vec<Ret, N> r;                                                       \
     detail::helper<N - 1>().run_1v_2p(                                         \
         r, [](Arg1 x, Arg2 *y) { return cl::__host_std::Fun(x, y); }, x, y);   \
     return r;                                                                  \
   }
 
 #define __MAKE_1V_2V_3P(Fun, N, Ret, Arg1, Arg2, Arg3)                         \
-  __SYCL_EXPORT Ret##N Fun __NOEXC(Arg1##N x, Arg2##N y, Arg3##N *z) {         \
-    Ret##N r;                                                                  \
+  __SYCL_EXPORT sycl::vec<Ret, N> Fun __NOEXC(                                 \
```

Comment:
It is for remquo intrinsic. Why do you think it is __NOEXC ?
std::remquo may raise FE_INVALID in some cases (https://en.cppreference.com/w/cpp/numeric/math/remquo), or is it somewhat different?

Reply:
Sorry, you did not change the attributes here. My mistake. I am removing the comment.